src/physics/em/detail/RelativisticBremDXsection.i.hh

```diff
@@ -0,0 +1,211 @@
+    real_type dxsec = term * elem_data_.zFactor1 + onemy * elem_data_.zFactor2;
+
+    return max(dxsec, 0.0);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Compute screen_functions: Tsai's analytical approximations of coherent and
+ * incoherent screening function to the numerical screening functions computed
+ * by using the Thomas-Fermi model: Y.-S.Tsai, Rev. Mod. Phys. 49 (1977) 421.
+ */
+auto RelativisticBremDXsection::compute_screen_functions(real_type gam,
+                                                         real_type eps)
+    -> ScreenFunctions
+{
+    ScreenFunctions func;
+    real_type       gam2 = ipow<2>(gam);
+    real_type       eps2 = ipow<2>(eps);
+
+    func.phi1 = 16.863 - 2 * std::log(1 + 0.311877 * gam2)
+                + 2.4 * std::exp(-0.9 * gam) + 1.6 * std::exp(-1.5 * gam);
+    func.phi2 = 2 / (3 + 19.5 * gam + 18 * gam2);
+
+    func.psi1 = 24.34 - 2 * std::log(1 + 13.111641 * eps2)
+                + 2.8 * std::exp(-8 * eps) + 1.2 * std::exp(-29.2 * eps);
+    func.psi2 = 2 / (3 + 120 * eps + 1200 * eps2);
+
+    return func;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Compute the LPM fsuppression unctions.
+ */
+auto RelativisticBremDXsection::compute_lpm_functions(real_type egamma)
+    -> LPMFunctions
+{
+    LPMFunctions func;
+
+    // Calcuate s
+    real_type y = egamma / total_energy_;
+
+    real_type sprime
+        = std::sqrt(0.125 * y * lpm_energy_ / ((1 - y) * total_energy_));
+
+    real_type xi_sprime{2};
+
+    if (sprime > 1)
+    {
+        xi_sprime = 1;
+    }
+    else if (sprime > constants::sqrt_two * elem_data_.s1)
+    {
+        real_type inv_logs2 = elem_data_.inv_logs2;
+        real_type h_sprime  = std::log(sprime) * inv_logs2;
+        xi_sprime           = 1 + h_sprime
+                    - 0.08 * (1 - h_sprime) * h_sprime * (2 - h_sprime)
+                          * inv_logs2;
+    }
+
+    real_type s = sprime / std::sqrt(xi_sprime);
+
+    // Include a dielectric suppression effect into s according to Migdal
+    real_type shat = s * (1 + density_corr_ / ipow<2>(egamma));
+
+    // Calculate xi(s)
+
+    func.xis = 2;
+    if (shat > 1)
+    {
+        func.xis = 1;
+    }
+    else if (shat > elem_data_.s1)
+    {
+        func.xis = 1 + std::log(shat) * elem_data_.inv_logs1;
+    }
+
+    // Evaluate G(s) and phi(s): G4eBremsstrahlungRelModel::GetLPMFunctions
+    if (shat < shared_.limit_s_lpm())
+    {
+        real_type    val  = shat * shared_.inv_delta_lpm();
+        unsigned int ilow = static_cast<unsigned int>(val);
+        val -= ilow;
+
+        real_type xlow = shared_.lpm_table[ItemIdT{ilow}].Gs;
+        func.gs = xlow + (shared_.lpm_table[ItemIdT{ilow + 1}].Gs - xlow) * val;
+
+        xlow      = shared_.lpm_table[ItemIdT{ilow}].phis;
+        func.phis = xlow
+                    + (shared_.lpm_table[ItemIdT{ilow + 1}].phis - xlow) * val;
```

Comment:
Waiting for Seth's comment if any.

Reply:
Yeah, in the `physics/grid` directory there's the `Interpolator` class for general interpolations, and `UniformGrid` for safely finding indices on a uniform grid. This instance has the special case of `front = 0`, and it reuses several of the data. Since it's so special, let's not worry about reusing that code, but let's add some assertions and lambdas (one to do the interpolation, one to make the table access less ugly):
```suggestion
        CELER_ASSERT(shat >= 0)
        size_type ilow = static_cast<size_type>(shat);
        val -= ilow;
        CELER_ASSERT(ilow + 1 < shared_.lpm_table.size());

        auto linterp = [val](real_type lo, real_type hi) {
            return lo + (hi - lo) * val;
        };
        auto get_table = [ilow, this](size_type offset) {
            return this->shared_.lpm_table[ItemIdT{ilow + offset}];
        };
        func.gs   = linterp(get_table(0).gs, get_table(1).gs);
        func.phis = linterp(get_table(0).phis, get_table(1).phis);
```
The compiler should have no trouble optimizing it to identical code as you'd written.

Reply:
Thanks for the suggested change.  Updated accordingly (except that shat -> val in the first two lines).